header/p2p/server.go

```diff
@@ -69,17 +68,52 @@ func (serv *ExchangeServer) requestHandler(stream network.Stream) {
 	if err = stream.CloseRead(); err != nil {
 		log.Error(err)
 	}
+
+	var headers []*header.ExtendedHeader
 	// retrieve and write ExtendedHeaders
 	switch pbreq.Data.(type) {
 	case *p2p_pb.ExtendedHeaderRequest_Hash:
-		serv.handleRequestByHash(pbreq.GetHash(), stream)
+		headers, err = serv.handleRequestByHash(pbreq.GetHash())
 	case *p2p_pb.ExtendedHeaderRequest_Origin:
-		serv.handleRequest(pbreq.GetOrigin(), pbreq.GetOrigin()+pbreq.Amount, stream)
+		headers, err = serv.handleRequest(pbreq.GetOrigin(), pbreq.GetOrigin()+pbreq.Amount)
 	default:
 		log.Error("server: invalid data type received")
 		stream.Reset() //nolint:errcheck
 		return
 	}
+	var code p2p_pb.StatusCode
+	switch err {
+	case nil:
+		code = p2p_pb.StatusCode_OK
+	case header.ErrNotFound:
+		headers = make([]*header.ExtendedHeader, 1)
+		code = p2p_pb.StatusCode_NOT_FOUND
+	default:
+		stream.Reset() //nolint:errcheck
+		return
+	}
+
+	// write all headers to stream
+	for _, h := range headers {
+		if err := stream.SetWriteDeadline(time.Now().Add(writeDeadline)); err != nil {
+			log.Debugf("error setting deadline: %s", err)
+		}
+		var bin []byte
+		if h != nil {
```

Comment:
I'm not familiar with this code, so probably a silly question: What is the case when we send empty response with `statusOk` , e.g. `h==nil` situation? How is it different from `header.ErrNotFound`? Would be nice to have a comment for it.

Reply:
Another question I have whether we should use return an error in situation where is partial data is returned. Maybe add another `statusCodePartial` to keep status explicit?

Reply:
Good question. Could you please add it [here](https://github.com/celestiaorg/celestia-node/issues/1163) ?